### PR TITLE
Flac parser

### PR DESCRIPTION
**Pull request by linuss**
Implement a simple Flac parser. Includes tests and fixtures.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -101,7 +101,7 @@ class MyParser
   def call(io)
     # ... do some parsing with `io`
     magic_bytes = io.read(4)
-    return unless magic_bytes != 'XBMP'
+    return unless magic_bytes == 'XBMP'
```
[the remark this answers is not in the document]
Comment (linuss) [reply README.md]: Haha, thanks :) Copied my first version from the readme and it took me a while to figure out what went wrong

File: .gitignore [hunk 2]
```diff
@@ -10,4 +10,7 @@
 *.gem
 
 # rspec failure tracking
-.rspec_status
\ No newline at end of file
+.rspec_status
+
+# Vim temp-files
```
[the remark this answers is not in the document]
Comment (linuss) [reply .gitignore]: You're right, I've removed it!

**Comment by linuss** [comment]
Hey @WJWH, thanks for your review! I agree on all points, and have just pushed some commits that should address them.

**Comment by bermannoah** [comment]
Hey @linuss -- @WJWH is on holiday so I'm popping in to follow up -- this looks really good! Only one thing I might request is that at [line 56](https://github.com/WeTransfer/format_parser/pull/78/files#diff-d473fc272437f1cc29c325b3fefc980dR56) you also include a check for some Float oddities -- we've run into `Float::INFINITY` showing up a couple times out of the MP3 parser.

**Comment by linuss** [comment]
Hey @bermannoah I'm afraid I can't find what you're referring to in the MP3 parser code. I've done some googeling, and all I encountered was that 
```
1.0 / 0    #=> Infinity
1 / 0.0    #=> Infinity
```
(found here http://nithinbekal.com/posts/ruby-infinity/).
I think my code already handles both cases, as the right-hand side of this expression cannot be zero (because it is checked on line 42), but please tell me if you think that there is another way in which this could blow up! :)

**Comment by bermannoah** [comment]
Hey @linuss fair point! It occurs to me that we handled the 1.0/0 => Infinity thing elsewhere outside of format_parser 😅so all looks good to me.

**Pull request merged**